src/plugins/auto_batch/src/compiled_model.cpp

```diff
@@ -39,63 +40,38 @@ CompiledModel::~CompiledModel() {
     m_worker_requests.clear();
 }
 
-unsigned int CompiledModel::ParseTimeoutValue(const std::string& s) {
-    auto val = std::stoi(s);
-    if (val < 0)
-        IE_THROW(ParameterMismatch) << "Value for the " << CONFIG_KEY(AUTO_BATCH_TIMEOUT) << " should be unsigned int";
-    return val;
-}
-
-std::shared_ptr<InferenceEngine::RemoteContext> CompiledModel::GetContext() const {
-    return m_model_without_batch->GetContext();
-}
-
-InferenceEngine::IInferRequestInternal::Ptr CompiledModel::CreateInferRequestImpl(
-    InferenceEngine::InputsDataMap networkInputs,
-    InferenceEngine::OutputsDataMap networkOutputs) {
+std::shared_ptr<ov::ISyncInferRequest> CompiledModel::create_sync_infer_request() const {
     auto workerRequestPtrAndId = GetWorkerInferRequest();
-    return std::make_shared<SyncInferRequest>(networkInputs,
-                                              networkOutputs,
-                                              workerRequestPtrAndId.first,
-                                              workerRequestPtrAndId.second,
-                                              m_device_info.batch_for_device,
-                                              m_batched_inputs,
-                                              m_batched_outputs);
+    auto async_infer_request = std::make_shared<ov::autobatch_plugin::SyncInferRequest>(
+        std::static_pointer_cast<const ov::autobatch_plugin::CompiledModel>(shared_from_this()),
+        workerRequestPtrAndId.first,
+        workerRequestPtrAndId.second,
+        m_device_info.device_batch_size,
+        m_batched_inputs,
+        m_batched_outputs);
+    return async_infer_request;
```

Comment:
Can it be more useful to define `SyncInferRequest` as friend of `CompiledModel` in order to get `m_device_info`, `m_batched_inputs` and `m_batched_outputs` from compiled_model directly in `SyncInferRequest()` and decrease number of arguments?

Reply:
@vurusovs I think this interface for SyncInferRequest of batch is not only called by CompiledModel, but also can be called by others, in this situation the batchsize/inputs/outputs need to be parameter. So I think we need to keep this.